src/propslib/DelegationConfig.cpp

```diff
@@ -261,90 +223,62 @@ CATCH_RETURN()
-
-    return hr;
-}
-
-[[nodiscard]] HRESULT DelegationConfig::s_GetDefaultTerminalId(IID& iid) noexcept
-{
-    iid = { 0 };
-
-    auto hr = s_Get(DELEGATION_TERMINAL_KEY_NAME, iid);
 
-    auto defApp = false;
-    if (SUCCEEDED(Microsoft::Console::Internal::DefaultApp::CheckShouldTerminalBeDefault(defApp)) && defApp)
-    {
-        if (FAILED(hr))
+        if (bytesUsed == sizeof(buffer))
         {
-            // If we can't find a user-defined delegation console/terminal, use the hardcoded
-            // delegation console/terminal instead.
-            iid = CLSID_SystemDelegationTerminal;
-            hr = S_OK;
+            // RegQueryValueExW docs:
+            // If the data has the REG_SZ, REG_MULTI_SZ or REG_EXPAND_SZ type, the string may not have been stored with the
+            // proper terminating null characters. Therefore, even if the function returns ERROR_SUCCESS, the application
+            // should ensure that the string is properly terminated before using it; otherwise, it may overwrite a buffer.
+            buffer[std::size(buffer) - 1] = 0;
+            LOG_IF_FAILED(IIDFromString(&buffer[0], &values[i]));
         }
     }
 
-    return hr;
-}
-
-[[nodiscard]] HRESULT DelegationConfig::s_Get(PCWSTR value, IID& iid) noexcept
-{
-    wil::unique_hkey currentUserKey;
-    wil::unique_hkey consoleKey;
-
-    RETURN_IF_NTSTATUS_FAILED(RegistrySerialization::s_OpenConsoleKey(&currentUserKey, &consoleKey));
-
-    wil::unique_hkey startupKey;
-    RETURN_IF_NTSTATUS_FAILED(RegistrySerialization::s_OpenKey(consoleKey.get(), L"%%Startup", &startupKey));
-
-    DWORD bytesNeeded = 0;
-    auto result = RegistrySerialization::s_QueryValue(startupKey.get(),
-                                                      value,
-                                                      0,
-                                                      REG_SZ,
-                                                      nullptr,
-                                                      &bytesNeeded);
-
-    if (NTSTATUS_FROM_WIN32(ERROR_SUCCESS) != result)
+    if (values[0] == CLSID_Default || values[1] == CLSID_Default)
     {
-        RETURN_NTSTATUS(result);
+        return DefaultDelegationPair;
     }
-
-    auto buffer = std::make_unique<wchar_t[]>(bytesNeeded / sizeof(wchar_t) + 1);
-
-    DWORD bytesUsed = 0;
-
-    RETURN_IF_NTSTATUS_FAILED(RegistrySerialization::s_QueryValue(startupKey.get(),
-                                                                  value,
-                                                                  bytesNeeded,
-                                                                  REG_SZ,
-                                                                  reinterpret_cast<BYTE*>(buffer.get()),
-                                                                  &bytesUsed));
-
-    RETURN_IF_FAILED(IIDFromString(buffer.get(), &iid));
-
-    return S_OK;
+    if (values[0] == CLSID_Conhost || values[1] == CLSID_Conhost)

```

Comment:
Was the saved value SPECIFICALLY conhost? then return that



oh ho ho so I see, this is where we're being tricky now. Now conhost isn't {0}, it's got it's own GUID now. That makes sense.